server/src/main/scala/org/http4s/server/middleware/HttpMethodOverrider.scala

```diff
@@ -0,0 +1,87 @@
+package org.http4s.server.middleware
+
+import cats.Applicative
+import cats.data.Kleisli
+import cats.implicits._
+import org.http4s.util.CaseInsensitiveString
+import org.http4s.{AttributeKey, Header, HttpApp, Method, ParseResult, Request, Response, Status}
+
+object HttpMethodOverrider {
+
+  /**
+    * HttpMethodOverrider middleware config options.
+    */
+  final case class HttpMethodOverriderConfig(
+      overrideStrategy: OverrideStrategy,
+      overridableMethods: List[Method])
+
+  sealed trait OverrideStrategy
+  final case class HeaderOverrideStrategy(headerName: CaseInsensitiveString)
+      extends OverrideStrategy
+  final case class QueryOverrideStrategy(paramName: String) extends OverrideStrategy
+
+  val defaultConfig = HttpMethodOverriderConfig(
+    HeaderOverrideStrategy(CaseInsensitiveString("X-HTTP-Method-Override")),
+    List(Method.POST))
+
+  val overriddenMethodAttrKey: AttributeKey[Method] = AttributeKey[Method]
+
+  /** Simple middleware for HTTP Method Override.
+    *
+    * This middleware lets you use  HTTP verbs such as PUT or DELETE in places where the client
+    * doesn't support it. Camouflage your request with another HTTP verb(usually POST) and sneak
+    * the desired one using a custom header or request parameter. The middleware will '''override'''
+    * the original verb with the new one for you, allowing the request the be dispatched properly.
+    *
+    * @param http [[HttpApp]] to transform
+    * @param config http method overrider config
+    */
+  def apply[F[_]](http: HttpApp[F], config: HttpMethodOverriderConfig)(
+      implicit F: Applicative[F]): HttpApp[F] = {
+
+    def processRequestWithMethod(
+        req: Request[F],
+        parseResult: ParseResult[Method]): F[Response[F]] = parseResult match {
+      case Left(_) => F.pure(Response[F](Status.MethodNotAllowed))
```

Comment:
I think this would be a `BadRequest`.  `MethodNotAllowed` means we recognize the method.